### DISCUSSION TITLE
Training Custom Dataset . How to Identify my best more accurate model ??

### DISCUSSION THREAD

**Discussion by DawaraGit** [ultralytics/yolov5#12179] [Q&A]
Im running training on a custom DataSet with yolov5l6 as checkpoint.

I build my DataSet1 , DataSet2 and so on looking for my best results ( adding images, adding backgrounds , etc )

After each training session I run an inference to some test datasets and Visually ( with my eyes ) I analyze witch one is more accurate depending in what I want to detect.
Said that,
What is the right tool/script or steps to determine witch of my trained models are the best one ? 
And... after more training if my new models are better that the old ones .

thanks in advance !!!

**Comment by danielzhangau** [comment]
1. **Create an Isolated Test Set**: Make sure this dataset is diverse, representative of real-world scenarios, and doesn't overlap with your training or validation sets.
2. **Run Inference on the Isolated Test Set**: For each trained model, run inference on this test set. YOLOv5 should provide you with mAP, Precision, and Recall values.
